lib/SlackGhost.js

```diff
@@ -60,34 +60,49 @@ SlackGhost.prototype.update = function(message, room) {
     ]);
 };
 
-SlackGhost.prototype.updateDisplayname = function(message, room) {
+SlackGhost.prototype.getDisplayName = async function(message, room) {
+    const token = room.getAccessToken();
+    if (message.bot_id) {
+        const bot = await this.lookupBotInfo(message.bot_id, token);
+        if (bot && bot.name) {
+            return bot.name;
+        }
+        else {
+            return message.bot_id;
+        }
+    }
+    else {
+        const user = await this.lookupUserInfo(message.user_id, token);
+        if (user && user.profile) {
+            return user.profile.display_name || user.profile.real_name;
+        }
+    }
+    return null;
+};
+
+SlackGhost.prototype.updateDisplayname = async function(message, room) {
     var display_name = message.user_name;
 
-    var getDisplayName;
     if (!display_name) {
-        getDisplayName = this.lookupUserInfo(message.user_id, room.getAccessToken())
-            .then(user => {
-                if (user && user.profile) {
-                    return user.profile.display_name || user.profile.real_name;
-                }
-            });
-    } else {
-        getDisplayName = Promise.resolve(display_name);
-    }
+        const newDisplayName = await this.getDisplayName(message, room);
 
-    return getDisplayName.then(display_name => {
-        if (!display_name || this._display_name === display_name) return Promise.resolve();
+        display_name = newDisplayName;
 
         return this.getIntent().setDisplayName(display_name).then(() => {
             this._display_name = display_name;
             return this._main.putUserToStore(this);
         });
-    })
+
+    }
+    return;
 };
 
 SlackGhost.prototype.lookupAvatarUrl = function(user_id, token) {
     return this.lookupUserInfo(user_id, token).then((user) => {
-        if (!user || !user.profile) return;
+        if (!user || !user.profile) {
+            log.debug("Can't read user profile");
```

Comment:
```suggestion
            log.debug("Can't read user profile, so cannot lookup avatar url");
```

would be more helpful for debugging